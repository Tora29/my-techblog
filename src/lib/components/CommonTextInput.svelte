<script lang="ts">
	// コンポーネントの変数
	export let placeholder: string = 'Input here';
	export let style: string = 'input-bordered';
	export let color: string = 'input-primary';
	export let size: string = 'input-md';
</script>

<input type="text" {placeholder} class="input {style} {color} {size}" />
