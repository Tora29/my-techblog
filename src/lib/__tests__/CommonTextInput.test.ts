import { getByRole, render } from '@testing-library/svelte';
import { expect, test } from 'vitest';
import '@testing-library/jest-dom';
import CommonTextInput from '$lib/components/CommonTextInput.svelte';

describe('TextInput Component', () => {
	test('インプットエリアをデフォルトプロパティでレンダリング', () => {
		const { getByPlaceholderText, container } = render(CommonTextInput);

		// プレイスホルダーのデフォルト値を確認
		expect(getByPlaceholderText('Input here')).toBeInTheDocument();

		// スナップショットを取得
		expect(container).toMatchSnapshot();
	});

	test('インプットエリアをカスタムプロパティでレンダリング', () => {
		const props = {
			placeholder: 'test-placeholder',
			style: 'test-style',
			color: 'test-color',
			size: 'test-size'
		};

		const { getByPlaceholderText, getByRole, container } = render(CommonTextInput, { props });

		// カスタムプレイスホルダーの値を確認
		expect(getByPlaceholderText(props.placeholder)).toBeInTheDocument();

		// カスタムクラスを確認
		const input = getByRole('textbox');
		expect(input).toHaveClass('input', props.style, props.color, props.size);

		// スナップショットを取得
		expect(container).toMatchSnapshot();
	});
});
