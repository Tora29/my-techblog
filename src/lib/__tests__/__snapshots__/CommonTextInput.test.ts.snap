// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TextInput Component > インプットエリアをカスタムプロパティでレンダリング 1`] = `
<div>
  <input
    class="input test-style test-color test-size"
    placeholder="test-placeholder"
    type="text"
  />
</div>
`;

exports[`TextInput Component > インプットエリアをデフォルトプロパティでレンダリング 1`] = `
<div>
  <input
    class="input input-bordered input-primary input-md"
    placeholder="Input here"
    type="text"
  />
</div>
`;
